x-pack/plugins/security_solution/server/endpoint/services/agent/clients/endpoint/endpoint_agent_status_client.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { type AgentStatusRecords, HostStatus } from '../../../../../../common/endpoint/types';
+import type { ResponseActionAgentType } from '../../../../../../common/endpoint/service/response_actions/constants';
+import { AgentStatusClient } from '../lib/base_agent_status_client';
+import { getPendingActionsSummary } from '../../../actions';
+import { AgentStatusClientError } from '../errors';
+
+export class EndpointAgentStatusClient extends AgentStatusClient {
+  protected readonly agentType: ResponseActionAgentType = 'endpoint';
+
+  async getAgentStatuses(agentIds: string[]): Promise<AgentStatusRecords> {
+    const metadataService = this.options.endpointService.getEndpointMetadataService();
+    const esClient = this.options.esClient;
+
+    try {
+      const hostInfoForAgents = await Promise.all(
+        agentIds.map((agentId) =>
```

Comment:
getting one at a time feels expensive.  The metadata service has a `getMetadataForEndpoints()` method that accepts multiple IDs - can that be used? or maybe the the `getHostMetadataList()` using a `kuery`?

Reply:
That would be great. Unfortunately host metadata doesn't have `last_checkin` and `host_status` on it 😞

Reply:
I'm pretty sure `getHostMetadataList()` does sine it calls the enrich method internally. Just wondering if it would be more efficient - I also don't recall now what the `max` is for the number of agents IDs is for the API that calls this service

Reply:
hmm `getHostMetadataList` does have all `HostInfo`. Looks like it's never been used anywhere. It also needs an SoClient as param in addition to `kuery`. I'll give that a try

Reply:
I'm not sure I follow... That client is in use (called from the metadata route API handler) and should have everything it needs to get Endpoint metadata

Reply:
done 6d789d0d39feb1adc3c3a6514c2f03ff86de9763